File: Practica General Auxiliatura/POLIMORFISMO/Eje1/Eje1.java
class Videojuego {
    String nombre;
    String plataforma;
    int cantidadjugadores;

    public Videojuego(String nombre, String plataforma, int cantidadjugadores) {
        this.nombre = nombre;
        this.plataforma = plataforma;
        this.cantidadjugadores = cantidadjugadores;
    }

    public void mostrar() {
        System.out.println("Nombre del videojuego: " + this.nombre);
        System.out.println("Plataforma: " + this.plataforma);
        System.out.println("Cantidad de jugadores: " + this.cantidadjugadores);
    }

    public void agregarJugadores(int cantidad) {
        this.cantidadjugadores += cantidad;
        System.out.println("Ahora el videojuego " + this.nombre + " tiene " + this.cantidadjugadores + " jugadores.");
    }

    public static void main(String[] args) {
        Videojuego videojuego1 = new Videojuego("FIFA 23", "PlayStation 5", 2);
        Videojuego videojuego2 = new Videojuego("Minecraft", "PC", 1);

        videojuego1.mostrar();
        videojuego2.mostrar();

        videojuego1.agregarJugadores(1);
        videojuego2.agregarJugadores(3);

        videojuego1.mostrar();
        videojuego2.mostrar();
    }
}

File: Practica General Auxiliatura/POLIMORFISMO/Eje1/Eje1.py
class Videojuego:
    def __init__(self, nombre, plataforma, cantidadjugadores=1):
        self.nombre = nombre
        self.plataforma = plataforma
        self.cantidadjugadores = cantidadjugadores

    def mostrar(self):
        print(f"Nombre del videojuego: {self.nombre}")
        print(f"Plataforma: {self.plataforma}")
        print(f"Cantidad de jugadores: {self.cantidadjugadores}")

    def agregarJugadores(self, cantidad=1):
        self.cantidadjugadores += cantidad
        print(f"Ahora el videojuego {self.nombre} tiene {self.cantidadjugadores} jugadores.")

videojuego1 = Videojuego("FIFA 23", "PlayStation 5", 2)
videojuego2 = Videojuego("Minecraft", "PC", 1)

videojuego1.mostrar()
videojuego2.mostrar()

videojuego1.agregarJugadores()
videojuego2.agregarJugadores(3)

videojuego1.mostrar()
videojuego2.mostrar()
